### DISCUSSION TITLE
Will Chisel add support for Scala3 ? (Chisel是否会加入对Scala3的支持?)

### DISCUSSION THREAD

**Discussion by SDTEFU** [chipsalliance/chisel#2939] [Q&A]
Will Chisel add support for Scala3 ?                 (Chisel是否会加入对Scala3的支持?)
How to use Scala3 to call Chisel3 normally ?  (如何使用Scala3来正常调用Chisel3)

**Comment by seldridge** [comment]
There is an ongoing effort to do this lead by @adkian-sifive. Pieces have been migrated to Scala3 (Scala code which works with Scala2 or Scala3). However, there are a number of difficult parts stemming from use of macros and compiler plugins. The current goal is to get all tests running in both Scala2 and Scala3.
